standalone/pruntime/enclave/src/contracts/btc_lottery.rs

```diff
@@ -0,0 +1,451 @@
+    Error(Error),
+}
+
+impl BtcLottery {
+    /// Initializes the contract
+    pub fn new(secret: Option<ecdsa::Pair>) -> Self {
+        let token_set = BTreeMap::<u32, Vec<String>>::new();
+        let lottery_set = BTreeMap::<u32, BTreeMap<String, PrivateKey>>::new();
+        let utxo = BTreeMap::<u32, BTreeMap<Address, (Txid, u32, u64)>>::new();
+        let admin = AccountIdWrapper::from_hex(ALICE);
+        BtcLottery {
+            round_id: 0,
+            token_set,
+            lottery_set,
+            sequence: 0,
+            queue: Vec::new(),
+            secret,
+            utxo,
+            admin,
+        }
+    }
+
+    pub fn sign(
+        secp: &Secp256k1<All>,
+        digest: &[u8],
+        key: &PrivateKey,
+    ) -> Result<Signature, bitcoin::secp256k1::Error> {
+        Ok(secp.sign(&Message::from_slice(digest)?, &key.key))
+    }
+
+    pub fn new_round(&mut self, round_id: u32, total_count: u32, winner_count: u32) {
+        if !self.token_set.contains_key(&round_id) && !self.lottery_set.contains_key(&round_id) {
+            let secret = match self.secret.as_ref() {
+                Some(s) => s.to_raw_vec(),
+                None => {
+                    error!("LotteryNewRound: empty secret key");
+                    return;
+                }
+            };
+            let token_id: U256 = U256::from(round_id) << 128;
+            let mut round_token = Vec::new();
+            for index_id in 1..=total_count {
+                let nft_id = (token_id + index_id) | *TYPE_NF_BIT;
+                let token_id = format!("{:#x}", nft_id);
+                round_token.push(token_id);
+            }
+            let mut lottery_token = BTreeMap::<String, PrivateKey>::new();
+            let raw_seed = blake2_256(&Encode::encode(&(secret, round_id)));
+            let mut r: StdRng = SeedableRng::from_seed(raw_seed.clone());
+            let sample = round_token
+                .iter()
+                .choose_multiple(&mut r, winner_count as usize);
+            let mut salt = round_id * 10000;
+            for winner_id in sample {
+                let s = Secp256k1::new();
+                let raw_data = (raw_seed.clone(), salt);
+                let seed = blake2_256(&Encode::encode(&raw_data));
+                let sk = match ExtendedPrivKey::new_master(Network::Bitcoin, &seed) {
+                    Ok(e) => e.private_key,
+                    Err(err) => {
+                        error!(
+                            "LotteryNewRound: cannot create a new secret key from the seed: {:?}",
+                            &seed
+                        );
+                        return;
+                    }
+                };
+                lottery_token.insert(String::from(winner_id), sk);
+                salt += 1;
+            }
+            self.lottery_set.insert(round_id, lottery_token);
+            self.token_set.insert(round_id, round_token);
+            self.round_id = round_id;
+        }
```

Comment:
Add a warning for `else` case (e.g. else { error!("Round {} has already started"); })

Reply:
Added `else` warning for both `NewRound` and `OpenLottery` events